/system/info endpoint returns nil fields on ARMHF

## Expected Behaviour
Using `faas-cli version` against an `armhf` gateway should not panic.  It should provide the expected information in the same way as x86_64 does.
```
$ faas version
  ___                   _____           ____
 / _ \ _ __   ___ _ __ |  ___|_ _  __ _/ ___|
| | | | '_ \ / _ \ '_ \| |_ / _` |/ _` \___ \
| |_| | |_) |  __/ | | |  _| (_| | (_| |___) |
 \___/| .__/ \___|_| |_|_|  \__,_|\__,_|____/
      |_|

CLI:
 commit:  5b36c50280d961d3aa248bd17f901f4ef2774447
 version: 0.8.1

Gateway
 uri:     http://127.0.0.1:8080
 version: 0.9.11
 sha:     78a4584019a2a5d86b606a00b4fba979930cc2a7
 commit:  Add sha256 hashgen for watchdog to ci config


Provider
 name:          faas-swarm
 orchestration: swarm
 version:       0.5.0 
 sha:           4269ab15673ad883bfdce4df47c055588e8e86b2
```

Accessing http://<OPENFAAS_URL>:8080/system/info returns:

```
{"provider":{"provider":"faas-swarm","version":{"sha":"4269ab15673ad883bfdce4df47c055588e8e86b2","release":"0.5.0"},"orchestration":"swarm"},"version":{"commit_message":"Add sha256 hashgen for watchdog to ci config","sha":"78a4584019a2a5d86b606a00b4fba979930cc2a7","release":"0.9.11"}}
```

## Current Behaviour

An ungraceful error is returned.

```
$ faas version
  ___                   _____           ____
 / _ \ _ __   ___ _ __ |  ___|_ _  __ _/ ___|
| | | | '_ \ / _ \ '_ \| |_ / _` |/ _` \___ \
| |_| | |_) |  __/ | | |  _| (_| | (_| |___) |
 \___/| .__/ \___|_| |_|_|  \__,_|\__,_|____/
      |_|

CLI:
 commit:  a141dedf94ffeed84412365fd591bdc8999c5a1b
 version: 0.8.3
panic: interface conversion: interface {} is nil, not string

goroutine 1 [running]:
github.com/openfaas/faas-cli/commands.getGatewayDetails(...)
	/go/src/github.com/openfaas/faas-cli/commands/version.go:120
github.com/openfaas/faas-cli/commands.printServerVersions()
	/go/src/github.com/openfaas/faas-cli/commands/version.go:78 +0x42c
github.com/openfaas/faas-cli/commands.runVersion(0x640d68, 0x653e90, 0x0, 0x0)
	/go/src/github.com/openfaas/faas-cli/commands/version.go:54 +0x144
github.com/openfaas/faas-cli/vendor/github.com/spf13/cobra.(*Command).execute(0x640d68, 0x653e90, 0x0, 0x0, 0x640d68, 0x653e90)
	/go/src/github.com/openfaas/faas-cli/vendor/github.com/spf13/cobra/command.go:766 +0x20c
github.com/openfaas/faas-cli/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x63e898, 0x0, 0x3fc780, 0x18e4b80)
	/go/src/github.com/openfaas/faas-cli/vendor/github.com/spf13/cobra/command.go:852 +0x210
github.com/openfaas/faas-cli/vendor/github.com/spf13/cobra.(*Command).Execute(0x63e898, 0x6, 0x388dd1)
	/go/src/github.com/openfaas/faas-cli/vendor/github.com/spf13/cobra/command.go:800 +0x1c
github.com/openfaas/faas-cli/commands.Execute(0x18f6250, 0x2, 0x2)
	/go/src/github.com/openfaas/faas-cli/commands/faas.go:77 +0x90
main.main()
	/go/src/github.com/openfaas/faas-cli/app.go:19 +0x64
```
Accessing http://<OPENFAAS_URL>:8080/system/info returns:

```
{"provider":{"provider":"faas-swarm","version":{"sha":"4269ab15673ad883bfdce4df47c055588e8e86b2","release":"0.5.0"},"orchestration":"swarm"},"version":{"sha":"","release":"dev"}}
```

## Possible Solution
The `/system/info/` endpoint is missing key information on `armhf`.  The contents of `Dockerfile.armhf` could be updated to more closely match that of `Dockerfile` including the build-args and license check at 0.2.3.

## Steps to Reproduce (for bugs)
1. Clone repo onto an ARM32 machine
2. Run `./deploy_stack.armhf.sh`
3. `curl -sSL https://cli.openfaas.com | sudo sh`
4. `faas version`

## Context
Inconsistent experience
Ugly error message

## Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in -->
* FaaS-CLI version ( Full output from: `faas-cli version` ):
```
$ faas version
  ___                   _____           ____
 / _ \ _ __   ___ _ __ |  ___|_ _  __ _/ ___|
| | | | '_ \ / _ \ '_ \| |_ / _` |/ _` \___ \
| |_| | |_) |  __/ | | |  _| (_| | (_| |___) |
 \___/| .__/ \___|_| |_|_|  \__,_|\__,_|____/
      |_|

CLI:
 commit:  a141dedf94ffeed84412365fd591bdc8999c5a1b
 version: 0.8.3
panic: interface conversion: interface {} is nil, not string

goroutine 1 [running]:
github.com/openfaas/faas-cli/commands.getGatewayDetails(...)
	/go/src/github.com/openfaas/faas-cli/commands/version.go:120
github.com/openfaas/faas-cli/commands.printServerVersions()
	/go/src/github.com/openfaas/faas-cli/commands/version.go:78 +0x42c
github.com/openfaas/faas-cli/commands.runVersion(0x640d68, 0x653e90, 0x0, 0x0)
	/go/src/github.com/openfaas/faas-cli/commands/version.go:54 +0x144
github.com/openfaas/faas-cli/vendor/github.com/spf13/cobra.(*Command).execute(0x640d68, 0x653e90, 0x0, 0x0, 0x640d68, 0x653e90)
	/go/src/github.com/openfaas/faas-cli/vendor/github.com/spf13/cobra/command.go:766 +0x20c
github.com/openfaas/faas-cli/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x63e898, 0x0, 0x3fc780, 0x18e4b80)
	/go/src/github.com/openfaas/faas-cli/vendor/github.com/spf13/cobra/command.go:852 +0x210
github.com/openfaas/faas-cli/vendor/github.com/spf13/cobra.(*Command).Execute(0x63e898, 0x6, 0x388dd1)
	/go/src/github.com/openfaas/faas-cli/vendor/github.com/spf13/cobra/command.go:800 +0x1c
github.com/openfaas/faas-cli/commands.Execute(0x18f6250, 0x2, 0x2)
	/go/src/github.com/openfaas/faas-cli/commands/faas.go:77 +0x90
main.main()
	/go/src/github.com/openfaas/faas-cli/app.go:19 +0x64
```

* Docker version `docker version` (e.g. Docker 17.0.05 ):
```
 docker version
Client:
 Version:           18.09.0
 API version:       1.39
 Go version:        go1.10.4
 Git commit:        4d60db4
 Built:             Wed Nov  7 00:57:21 2018
 OS/Arch:           linux/arm
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          18.09.0
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.10.4
  Git commit:       4d60db4
  Built:            Wed Nov  7 00:17:57 2018
  OS/Arch:          linux/arm
  Experimental:     false
```
* Are you using Docker Swarm or Kubernetes (FaaS-netes)?
Swarm

* Operating System and version (e.g. Linux, Windows, MacOS):
Raspbian

--- a/gateway/Dockerfile.armhf
+++ b/gateway/Dockerfile.armhf
@@ -1,4 +1,11 @@
 FROM golang:1.10.4 as build
+ARG GIT_COMMIT_SHA
+ARG GIT_COMMIT_MESSAGE
+ARG VERSION='dev'
+
+RUN curl -sSfL https://github.com/alexellis/license-check/releases/download/0.2.3/license-check-armhf \
+    > /usr/bin/license-check \
+    && chmod +x /usr/bin/license-check
 
 WORKDIR /go/src/github.com/openfaas/faas/gateway
 
@@ -16,17 +23,39 @@ COPY version        version
 COPY scaling        scaling
 COPY server.go      .
 
-RUN GOARM=7 CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo -o gateway .
+# Run a gofmt and exclude all vendored code.
+RUN license-check -path ./ --verbose=false "Alex Ellis" "OpenFaaS Project" "OpenFaaS Authors" "OpenFaaS Author(s)" \
+    && test -z "$(gofmt -l $(find . -type f -name '*.go' -not -path "./vendor/*"))" \
+    && go test $(go list ./... | grep -v integration | grep -v /vendor/ | grep -v /template/) -cover \
+    && GOARM=7 CGO_ENABLED=0 GOOS=linux go build --ldflags "-s -w \
+    -X github.com/openfaas/faas/gateway/version.GitCommitSHA=${GIT_COMMIT_SHA}\
+    -X \"github.com/openfaas/faas/gateway/version.GitCommitMessage=${GIT_COMMIT_MESSAGE}\"\
+    -X github.com/openfaas/faas/gateway/version.Version=${VERSION}" \
+    -a -installsuffix cgo -o gateway .
 
 FROM alpine:3.8
-WORKDIR /root/
+
+LABEL org.label-schema.license="MIT" \
+    org.label-schema.vcs-url="https://github.com/openfaas/faas" \
+    org.label-schema.vcs-type="Git" \
+    org.label-schema.name="openfaas/faas" \
+    org.label-schema.vendor="openfaas" \
+    org.label-schema.docker.schema-version="1.0"
+
+RUN addgroup -S app \
+    && adduser -S -g app app
+
+WORKDIR /home/app
 
 EXPOSE 8080
 ENV http_proxy      ""
 ENV https_proxy     ""
 
-COPY --from=0 /go/src/github.com/openfaas/faas/gateway/gateway    .
+COPY --from=build /go/src/github.com/openfaas/faas/gateway/gateway    .
 
 COPY assets     assets
-RUN sed -ie s/store.json/store-armhf.json/g /root/assets/script/funcstore.js
+RUN sed -ie s/store.json/store-armhf.json/g /home/app/assets/script/funcstore.js
+RUN chown -R app:app ./
+USER app
+
 CMD ["./gateway"]

